_posts/2019-09-20-lambda-to-openfaas.md

```diff
@@ -0,0 +1,282 @@
+---
+title: "Migrate Your AWS Lambda Functions to OpenFaaS"
+description: Burton explains the steps for migrating an existing AWS Lambda function to OpenFaaS
+date: 2019-09-10
+image: /images/lambda-to-openfaas/birds.jpg
+categories:
+  - Lambda
+  - tutorial
+  - examples
+author_staff_member: burton
+dark_background: false
+---
+
+In this post, I'll walk you through the steps taken to migrate existing AWS Lambda functions to OpenFaaS compatible functions in order to allow easier local testing, familiar development process, and the freedom to use any number of other cloud providers without any code changes.
+
+AWS introduced the world to the idea of serverless architecture back in 2014 with the announcement of Lambda. The idea was that you could upload a zip file containing your code logic, and AWS would manage the infrastructure behind the scenes, billing you only for the time the code was being executed.
+
+In his talk [Welcome to Serverless 2.0](https://www.youtube.com/watch?v=JvXm-oHi5Mg) Alex Ellis described the new standard for serverless functions: Docker/OCI images exposing a port for execution, deployed to a Kubernetes cluster. This makes your functions completely portable and removes many of the limitations of the previous generation of serverless functions.
+
+Without further ado, let's see what it takes to upgrade your AWS Lambda functions to OpenFaaS and Serverless 2.0!
+
+## Prerequisites
+
+- Existing AWS Lambda functions (optional)
+  - For the tutorial, we'll be using the Lambda function in this [Github repository](https://github.com/burtonr/lambda-openfaas-blog)
+- OpenFaaS 
+  - Check the documentation for [how to deploy to Kubernetes](https://docs.openfaas.com/deployment/kubernetes/)
+- OpenFaaS CLI
+  - Check the documentation for [installation instructions](https://docs.openfaas.com/cli/install/)
+- IDE, or other text editor
+  - I recommend [Visual Studio Code](https://code.visualstudio.com/)
+
+## AWS Lambda
+
+The Lambda functions we'll be migrating are part of a URL shortener. One function creates and stores the shortened URL in DynamoDB, and the other function is responsible for looking up the destination URL, and redirecting the browser.
+
+Let's have a look at the `shortener` function which creates and stores the shortened URLs:
+
+```javascript
```

Comment:
I didn't look yet, but is this in a GitHub repo ?

Reply:
Yes, there is a link in the prerequisites mentioning "Existing AWS Lambda functions". 

https://github.com/burtonr/lambda-openfaas-blog

Reply:
Can you say what the filename is under each code snippet and make it a hyperlink?